src/overlays/actors/ovl_En_Aob_01/z_en_aob_01.c

```diff
@@ -488,124 +553,152 @@ void func_809C1EC8(EnAob01* this, PlayState* play) {
             if (this->collider.base.ocFlags2 & OC2_HIT_PLAYER) {
                 this->actor.flags |= ACTOR_FLAG_10000;
                 func_800B8614(&this->actor, play, 100.0f);
-                this->unk_2D2 |= 8;
-                this->actionFunc = func_809C21E0;
+                this->stateFlags |= ENAOB01_FLAG_TALKING_TO_PLAYER_HOLDING_DOG;
+                this->actionFunc = EnAob01_BeforeRace_Talk;
             }
-        } else if (Actor_ProcessTalkRequest(&this->actor, &play->state) && (this->unk_2D2 & 0x100)) {
-            this->unk_2D2 &= ~0x100;
-            this->unk_2E6 = this->trackTarget;
-            this->unk_2EC = this->headRot;
-            this->unk_2F2 = this->torsoRot;
-            func_809C16DC(this, play);
-            this->actionFunc = func_809C21E0;
+        } else if (Actor_ProcessTalkRequest(&this->actor, &play->state) &&
+                   (this->stateFlags & ENAOB01_PLAYER_CAN_TALK)) {
+            this->stateFlags &= ~ENAOB01_PLAYER_CAN_TALK;
+            this->tempTrackTarget = this->trackTarget;
+            this->tempHeadRot = this->headRot;
+            this->tempTorsoRot = this->torsoRot;
+            EnAob01_BeforeRace_HandleConversation(this, play);
+            this->actionFunc = EnAob01_BeforeRace_Talk;
         } else {
-            this->unk_2D2 &= ~0x100;
+            this->stateFlags &= ~ENAOB01_PLAYER_CAN_TALK;
             if ((this->actor.xzDistToPlayer < 100.0f) && !(this->collider.base.ocFlags2 & OC2_HIT_PLAYER)) {
-                this->unk_2D2 |= 0x100;
+                this->stateFlags |= ENAOB01_PLAYER_CAN_TALK;
                 func_800B8614(&this->actor, play, 100.0f);
             }
         }
     }
 }
 
-void func_809C21E0(EnAob01* this, PlayState* play) {
+/**
+ * Handles the actual action of talking with the player before the race begins.
+ */
+void EnAob01_BeforeRace_Talk(EnAob01* this, PlayState* play) {
     u8 talkState = Message_GetState(&play->msgCtx);
 
     Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 4, 4000, 1);
 
-    if ((this->unk_2D2 & 8) && !func_809C2EC4(this, play)) {
-        if ((this->unk_210 != 0) && (this->unk_210 != 0x3535) && (this->unk_210 != 0x3524) &&
-            (this->unk_210 != 0x3548) && (this->unk_210 != 0x3549) && (this->unk_210 != 0x354A)) {
-            this->unk_210 = 0x3523;
+    //! @bug: This block of code is *supposed* to act as a failsafe for when the player triggered this conversation by
+    //! bumping into the racetrack owner while holding the dog and, at the same time, dropped the dog frame-perfectly.
+    //! Without this code, doing those same actions will cause the ractrack owner be stuck in her "talk" action
+    //! permanently, so the developers probably put this code in to prevent it. However, this block of code, as-is,
+    //! causes a much worse problem; for some unknown reason, this code softlocks the game if the player manages to
+    //! perform the simultaneous actions described above.
```

Comment:
Perhaps it would be a good idea to mention what Zoey mentioned in discord.
https://discord.com/channels/688807550715560050/688851976746041391/1098248443291521044
TLDR. The code seems to work fine when you throw the dog, but if you shield drop it that is when there is an issue.

Reply:
Whoops, I was going to fix up this comment, but I forgot to actually do that before pushing. I'll update it, thanks for the reminder.